### PR TITLE
Add  williamfalcon as owner for API changes

### PR DESCRIPTION
**Pull request by edenlightning**
Add williamfalcon's approval for API changes.

### CODE REVIEW

**Comment by teddykoker** [comment]
Pretty sure this won't do anything as everyone still owns `*`

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -12,4 +12,9 @@
 /tests/metrics/*              @teddykoker @ananyahjha93 @justusschock
 /docs/source/metrics.rst      @teddykoker @ananyahjha93 @justusschock
 
-
+# API
+/pytorch_lightning/callbacks/base.py  @williamfalcon
+/pytorch_lightning/core/datamodule.py @williamfalcon
+/pytorch_lightning/trainer/trainer.py @williamfalcon
+/pytorch_lightning/core/hooks.py.     @williamfalcon
```
Comment (teddykoker) [inline .github/CODEOWNERS:19]:
```suggestion
/pytorch_lightning/core/hooks.py     @williamfalcon
```

**Review by teddykoker — CHANGES_REQUESTED** [review]

**Comment by edenlightning** [comment]
Screenshot - 2020-11-10T183323 507

then this doesn't make sense, no?

**Comment by williamFalcon** [comment]
doesn’t this also make it so that i have to approve in addition to the * people? for these files?

**Comment by edenlightning** [comment]
"The last matching pattern has the most precedence" (see [this](https://docs.github.com/en/enterprise-server@2.20/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax)).

I'm just wondering if it means it will assign **only** you to review. Not clear on that. Any idea?

**Review by Borda — COMMENTED** [review]
Will is already in all revies as default do what does this change shall add?
just a note, this also included all bug-fixed in theses files, not design/APi changes

**Comment by edenlightning** [comment]
@Borda do you know if we can make him required somehow only for these files? but not the only one?

**Comment by williamFalcon** [comment]
let's add this for now and figure out changes later.

the behavior that i would love is:
for certain files i need to be an additional approver.

ie: if person modified lightning.py, then the standard reviewers apply. In addition, my approval is required.

**Pull request merged**